Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1850,13 +1962,19 @@ var event = function(
         traverse(recMapping[key], recEvt[key], key);
       }
     };
-    argMapping.forEach((mapping, idx) => {
-      traverse(mapping, args[idx], 'arg' + idx);
-    });
-    if (config && config.listener) {
-      config.listener.apply(null, args);
-    }
-  };
+  }
+}
+
+var event = function(
+  argMapping: Array<?Mapping>,
+  config?: ?EventConfig,
+): any {
+  const animatedEvent = new AnimatedEvent(argMapping, config);
+  if (animatedEvent.__isNative) {
+    return animatedEvent;
```

Comment:
we should perhaps validate that argMapping has just `nativeEvent` key present (that's the data provided from the native part). There are many cases when you may want to use synthetic params (e.g. with pan responder) and in case someone try to set synthetic values mapping for natively driven value we should display a warning